pkg/controller/traceflow/controller.go

```diff
@@ -326,28 +335,41 @@ func (c *Controller) occupyTag(tf *opsv1alpha1.Traceflow) error {
 	return nil
 }
 
-func (c *Controller) allocateTag(tf *opsv1alpha1.Traceflow) (uint8, error) {
+// Allocates a tag. If the Traceflow request has been allocated with a tag
+// already, 0 is returned. If number of existing Traceflow requests reaches
+// the upper limit, an error is returned.
+func (c *Controller) allocateTag(name string) (uint8, error) {
 	c.runningTraceflowsMutex.Lock()
 	defer c.runningTraceflowsMutex.Unlock()
+
+	for _, n := range c.runningTraceflows {
+		if n == name {
+			// The Traceflow request has been processed already.
```

Comment:
I guess using name as the key might not be safe. For example, what will happen if controller disconnected from K8s apiserver, and then a processed CRD is deleted but a new CRD with the same name is created before controller connects to K8s apiserver?
But maybe ok let customer delete the CRD created during controller disconnected/down.

Reply:
Yes, to solve this case we need to store tf name and uuid into c.runningTraceflows, and improve allocate/deallocate logic.

Reply:
Right. Feel it is too big a change, and maybe not worthwhile.